### ISSUE TITLE
Unable to fetch deleted record using SOQL query ?

### ISSUE THREAD

**Issue by sachin-aryal** [simple-salesforce/simple-salesforce#197]
I was trying to query deleted record using SOQL query like this

SELECT Name, Id FROM Acocunt where IsDeleted = true and tried using the **queryAll** method, but this method actually just redirecting it to the **query** method.
Because of that the rest url was looking like this 
https://instance.salesforce.com/services/data/v40.0/query/  as you may know this directly exclude the deleted records. What i wanted was this url
https://instance.salesforce.com/services/data/v40.0/queryAll/.

So I think there needs some changes to support this type of case.

I tried this method sf.Account.deleted(start_time, end_time) but it only provide the record id and deletedDate.

**Comment by skamensky** [comment]
I agree with you. This is confusing. Since in simple-salesforce 'queryAll' is, according to the documentation, just a convenience 
https://github.com/simple-salesforce/simple-salesforce/blob/dda2708f13e815566f87f182601d23717797632b/simple_salesforce/api.py#L396.

And Salesforce's queryAll endpoint is used 

> to execute a SOQL query that includes information about records that have been deleted because of a merge or delete.


simple-salesforce doesn't have a convenience wrapper around Salesforce's queryAll endpoint. The solution is to use simple-salesforce's restful method which lets you hit any endpoint you want.

`sf.restful('queryAll',{'q' : 'SELECT Id,Name,LastModifiedDate,IsDeleted FROM Account WHERE IsDeleted=True LIMIT 2'})`

Either way, I think you're right in saying there needs to be some change. simple-salesforce's API method names shouldn't coincide with Salesforce's REST API endpoints unless they do the same thing.

Maybe we should rename `query_all` to `query_all_results`?

**Comment by sachin-aryal** [comment]
Yes you are right. I spent a lot of time searching for why I am not able to fetch deleted records until yesterday I found out the **queryAll** method actually not hitting the queryAll endpoint in salesforce.

**Comment by djdeejay** [comment]
+1 for renaming the query_all

+1 for adding a Parameter ALL which switches between query and queryAll endpoints

**Comment by jw207427** [comment]
I believe this is fixed.

query_all now has a flag to included deleted records.

query_all(query, include_deleted=False, **kwargs)

**Comment by jeffolsi** [comment]
Yes this is fixed

**Comment by sacses** [comment]
The flag naming is actually a little bit misleading. When the parameter is set to `True`, `queryAll` endpoint is called instead of `query`, but it doesn't necessarily mean that ONLY deleted records can be retrieved

> QueryAll also returns information about archived task and event records

Maybe renaming the flag to `include_all` or simply call the `queryAll` endpoint by default when using the `.query_all()` method would have been clearer.
